core/circuitbreaker/rule_manager.go

```diff
@@ -292,10 +293,16 @@ func onRuleUpdate(rules []*Rule) (err error) {
 		}
 	}
 
+	updateMux.Lock()
 	breakerRules = newBreakerRules
 	breakers = newBreakers
 	currentRules = rules
+	updateMux.Unlock()
 
+	logging.Debug("[CircuitBreaker onRuleUpdate] Time statistics(ns) for updating circuit breaker rule", "timeCost", util.CurrentTimeNano()-start)
+	updateMux.RLock()
```

Comment:
Maybe unnecessary?

Reply:
Above breakers = newBreakers.  So we shoule protect it

Reply:
We have the global ruleUpdateMux, here updateMux is unnecessary

Reply:
done